src/components/YouTube/YouTubeProvider.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { useContext, useReducer } from 'react';
+
+const YoutTubeContext = React.createContext();
+
+function youtubeReducer(state, action) {
+  switch (action.type) {
+    case 'search':
+      return {
+        ...state,
+        search: action.payload,
+      };
+    case 'currentVideo':
+      return {
+        ...state,
+        currentVideo: action.payload,
+      };
+    case 'closeCurrentVideo':
+      return {
+        ...state,
+        currentVideo: null,
+      };
+    case 'switchTheme':
+      return {
+        ...state,
+        theme: state.theme === 'dark' ? 'default' : 'dark',
+      };
+    default: {
+      return state;
```

Comment:
I'd throw an error here, it can prevent some unwanted stuff to go unnoticed, for example you can pass `CloseCurrentVideo` and the state won't update but it will be hard to find that the bug is caused because the `C` should be lower case like `closeCurrentVideo`, I learned it the hard way 😆